Hide internals in pytest stack traces

```
config = <gitman.models.config.Config object at 0x10b5c7b38>

    def it_should_fail_on_invalid_repositories(config):
        os.system("mkdir deps && touch deps/gitman_1")

        with pytest.raises(InvalidRepository):
>           gitman.lock()

/Users/Browning/Programs/gitman/tests/test_api.py:336:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/Users/Browning/Programs/gitman/gitman/commands.py:18: in wrapped
    result = func(*args, **kwargs)
/Users/Browning/Programs/gitman/gitman/commands.py:155: in lock
    count = config.lock_depenendencies(*names, obey_existing=False)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <gitman.models.config.Config object at 0x10b6bbf98>, args = ('lock_depenendencies',), kwargs = {}
mapper = <yorm.mapper.Mapper object at 0x10b6bbba8>

    @functools.wraps(method)
    def wrapped(self, *args, **kwargs):
        """Decorated method."""
        if not _private_call(method, args):
            mapper = common.get_mapper(self)
            if mapper and mapper.modified:
                log.debug("Loading before call: %s", method.__name__)
                mapper.load()
                if mapper.auto_save_after_load:
                    mapper.save()
                    mapper.modified = False

>       return method(self, *args, **kwargs)
E       AttributeError: 'Config' object has no attribute 'lock_depenendencies'

/Users/Browning/Programs/gitman/env/lib/python3.5/site-packages/YORM-1.1-py3.5.egg/yorm/bases/mappable.py:30: AttributeError
```

should look like:

```
config = <gitman.models.config.Config object at 0x10cd91d30>

    def it_should_fail_on_invalid_repositories(config):
        os.system("mkdir deps && touch deps/gitman_1")

        with pytest.raises(InvalidRepository):
>           gitman.lock()

/Users/Browning/Programs/gitman/tests/test_api.py:336:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/Users/Browning/Programs/gitman/gitman/commands.py:18: in wrapped
    result = func(*args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

root = '/private/tmp/gitman-shared', names = (), count = None
config = <gitman.models.config.Config object at 0x10cd95668>

    @restore_cwd
    def lock(*names, root=None):
        """Lock current dependency versions for a project.

        Optional arguments:

        - `*names`: optional list of dependency directory names to filter on
        - `root`: specifies the path to the root working tree

        """
        log.info("Locking dependencies...")
        count = None

        root = _find_root(root)
        config = load_config(root)

        if config:
            common.show()
            common.show("Locking dependencies...", color='message', log=False)
            common.show()
>           count = config.lock_depenendencies(*names, obey_existing=False)
E           AttributeError: 'Config' object has no attribute 'lock_depenendencies'

/Users/Browning/Programs/gitman/gitman/commands.py:155: AttributeError
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,8 +1,12 @@
 # Revision History
 
+## 1.2 (unreleased)
+
+- Updated base class to hide `pytest` traceback in wrapped methods.
+
 ## 1.1 (2016/10/22)
 
-- Add `data` property to `Mapper` as a hook for other serialization libraries.
+- Added `data` property to `Mapper` as a hook for other serialization libraries.
 
 ## 1.0.1 (2016/09/23)
 

--- a/yorm/__init__.py
+++ b/yorm/__init__.py
@@ -7,4 +7,4 @@ from .utilities import create, find, match, load, save, delete
 from .bases import Container, Converter, Mappable
 
 __project__ = 'YORM'
-__version__ = '1.1'
+__version__ = '1.2b1'

--- a/yorm/bases/mappable.py
+++ b/yorm/bases/mappable.py
@@ -18,6 +18,8 @@ def load_before(method):
     @functools.wraps(method)
     def wrapped(self, *args, **kwargs):
         """Decorated method."""
+        __tracebackhide__ = True  # pylint: disable=unused-variable
+
         if not _private_call(method, args):
             mapper = common.get_mapper(self)
             if mapper and mapper.modified:
@@ -43,6 +45,8 @@ def save_after(method):
     @functools.wraps(method)
     def wrapped(self, *args, **kwargs):
         """Decorated method."""
+        __tracebackhide__ = True  # pylint: disable=unused-variable
+
         result = method(self, *args, **kwargs)
 
         if not _private_call(method, args):

